integrations/access/accessrequest/app_test.go

```diff
@@ -100,7 +100,17 @@ func TestOpsGenieGetMessageRecipients(t *testing.T) {
 				},
 			}
 			recipients := a.getMessageRecipients(ctx, req)
-			require.ElementsMatch(t, tt.expectedRecipients, recipients)
+
+			got := make(map[string]string, len(recipients))
+			for _, r := range recipients {
+				got[r.Name] = r.ID
+			}
+			want := make(map[string]string, len(tt.expectedRecipients))
+			for _, r := range tt.expectedRecipients {
+				want[r.Name] = r.ID
+			}
+
+			require.Equal(t, want, got)
```

Comment:
If this fixes the issue, it's probably because we were getting duplicates in `recipients`.

We should look into why we get them, instead of ignoring them by using a `set`.

Reply:
The issue wasn't the duplicated recipients, it was that the test assumed an order to the return values. Added the results to a map just to compare without order mattering.

Reply:
According to the docs, `require.ElementsMatch` does not care about the order:

https://pkg.go.dev/github.com/stretchr/testify/require#ElementsMatch

> ElementsMatch asserts that the specified listA(array, slice...) is equal to specified listB(array, slice...) ignoring the order of the elements. If there are duplicate elements, the number of appearances of each of them in both lists should match.

Reply:
Ah sorry, you are right.
I am confused then by the message in the ticket then
`
expected: []common.Recipient{common.Recipient{Name:"foo", ID:"foo", Kind:"", Data:interface {}(nil)}, common.Recipient{Name:"bar", ID:"bar", Kind:"", Data:interface {}(nil)}}
actual  : []common.Recipient{common.Recipient{Name:"bar", ID:"bar", Kind:"", Data:interface {}(nil)}, common.Recipient{Name:"foo", ID:"foo", Kind:"", Data:interface {}(nil)}}
`
As it looks like the only difference is the order.

Reply:
🤔 
That error message comes from the fact that we **were** using `require.Equals`
However, it was fixed in this PR https://github.com/gravitational/teleport/pull/40777
This PR was created 4 days after the flaky test was reported (in April).

I think we can close this PR, unless I'm missing something